### PR TITLE
Fix nil pointer dereference in istiod-remote/templates/services.yaml

### PR DESCRIPTION
**Pull request by luksa**
The Helm values were flattened in #52036, but the endpoints.yaml and services.yaml templates weren't updated, causing the following error when rendering the istiod-remote chart:

```
executing "istiod-remote/templates/services.yaml" at <.Values.pilot.enabled>: nil pointer evaluating interface {}.enabled
```

### CODE REVIEW

**Review by zirain — APPROVED** [review]

File: manifests/charts/base/templates/endpoints.yaml [hunk 1]
```diff
@@ -3,7 +3,7 @@
 apiVersion: v1
 kind: Endpoints
 metadata:
-  {{- if .Values.pilot.enabled }}
```
Comment (howardjohn) [inline manifests/charts/base/templates/endpoints.yaml:6]: This doesn't make much sense to me. I would install with `helm install istio/base --set enabled=true`..? What does `enabled` mean -- nothing, without the context of `pilot.`

File: manifests/charts/istiod-remote/templates/endpoints.yaml [hunk 2]
```diff
@@ -3,7 +3,7 @@
 apiVersion: v1
 kind: Endpoints
 metadata:
-  {{- if .Values.pilot.enabled }}
+  {{- if .Values.enabled }}
```
Comment (howardjohn) [inline manifests/charts/istiod-remote/templates/endpoints.yaml:6]: How is this actually used? Does a user installing the remote chart actually decide between these two values, or did we just copy-paste from `base` and not change the chart to make sense?

Comment (nshankar13) [reply manifests/charts/istiod-remote/templates/endpoints.yaml:6]: This part seems copied over from the base chart - istiod-remote is effectively a combination of both base and istiod. Long-term, @bleggett and I were planning some additional changes to consolidate some of the installation params and trim some of the unnecessary components. I think for istiod-remote we can always assume that pilot.enabled is false (and configCluster is always true?), but right now we still need to explicitly set it.

Comment (bleggett) [reply manifests/charts/istiod-remote/templates/endpoints.yaml:6]: How `base` and `istiod-remote` are actually supposed to be used (and how people are supposed to know how to use them properly) in a multi-cluster+revisioned context is still a bit opaque to me.

File: manifests/charts/base/templates/endpoints.yaml (hunk 1 above)
Comment (bleggett) [reply manifests/charts/base/templates/endpoints.yaml:6]: I don't think this really needs to be in `base` anyway, and twiddled with it in https://github.com/istio/istio/pull/53016 - it is not a singleton, as it is revisioned.

File: manifests/charts/base/templates/services.yaml [hunk 3]
```diff
@@ -2,7 +2,7 @@
 apiVersion: v1
 kind: Service
 metadata:
-  {{- if .Values.pilot.enabled }}
```
Comment (bleggett) [inline manifests/charts/base/templates/services.yaml:5]: This also really doesn't belong in `base` - it is always revisioned and it should just go into `istio-discovery`, I think. I'll pick this back up in https://github.com/istio/istio/pull/53016

File: manifests/charts/base/templates/endpoints.yaml (hunk 1 above)
Comment (howardjohn) [reply manifests/charts/base/templates/endpoints.yaml:6]: @fjglira @luksa @bleggett @nshankar13 can you work together to come up with a plan to fix this up (before we ship it in 1.24)?

Comment (nshankar13) [reply manifests/charts/base/templates/endpoints.yaml:6]: I think the PR @bleggett linked somewhat addresses this, maybe we can continue the discussion there and decide how the values should be organized across the base, istio-discovery, and istiod-remote.

Comment (bleggett) [reply manifests/charts/base/templates/endpoints.yaml:6]: https://github.com/istio/istio/pull/53016 should resolve the concerns here.

**Pull request merged**